tools/logger/convert.c

```diff
@@ -284,6 +287,19 @@ static void process_params(struct proc_ldc_entry *pe,
 			p += 2;
 		}
 	}
+
+	/* ensure no NULLs in format strings */
+	for (j = 0; j < i; j++) {
+		if (!pe->params[j]) {
+			pe->params[j] = (uintptr_t)missing;
+			/* ensure object is no free()'ed */
+			pe->subst_mask &= ~(1 << j);
+		}
+
+		if (!pe->params[j])
+			abort();
```

Comment:
Actually L299-300 is not needed after I dropped the dynamic alloc of "<missing>". V4 coming...